lib/browser/api/browser-window.js

```diff
@@ -26,6 +27,33 @@ BrowserWindow.prototype._init = function () {
     ipcMain.emit('ELECTRON_GUEST_WINDOW_MANAGER_WINDOW_OPEN', event, url, frameName, disposition, options)
   })
 
+  this.webContents.on('-web-contents-created', (event, webContents, url,
+                                                frameName) => {
+    pendingWebContents.set(webContents, {url, frameName})
```

Comment:
We usually use `v8Util.setHiddenValue` to do this, so there is no need to worry about object's lifetime.